x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -165,7 +166,10 @@ export function toggleLayerVisible(layerId) {
 
 export function setSelectedLayer(layerId) {
   return async (dispatch, getState) => {
-
+    const transientLayerId = getTransientLayerId(getState());
```

Comment:
I'd create a new action-creator that wraps

```
const transientLayerId = getTransientLayerId(getState());
if (transientLayerId) {
      await dispatch(removeLayer(transientLayerId));
    }
```
Call this action-creator where needed so managing selected-layer state does not get tangled up with managing transient-layer state.

There's little reason to introduce new code after FF to solve hypothetical code-paths that don't exist (yes). (e.g. in what future scenario would fit-bounds govern selecting a layer?).

By entangling these, we are introducing the dependency where callers of `setSelectedLayer` use this function just to clear the transient-layer state. These are orthogonal concerns.